pkg/providers/tinkerbell/hardware/catalogue_hardware.go

```diff
@@ -1,21 +1,67 @@
 package hardware
 
 import (
-	"github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
+	"fmt"
+
+	tinkv1alpha1 "github.com/tinkerbell/tink/pkg/apis/core/v1alpha1"
 	corev1 "k8s.io/api/core/v1"
 	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
+	eksav1alpha1 "github.com/aws/eks-anywhere/pkg/api/v1alpha1"
 	"github.com/aws/eks-anywhere/pkg/constants"
 )
 
+type DiskExtractor struct {
+	Selector map[string]eksav1alpha1.HardwareSelector
+	Disks    map[string]string
+}
+
+func NewDiskExtractor() *DiskExtractor {
+	return &DiskExtractor{
+		Selector: make(map[string]eksav1alpha1.HardwareSelector),
+		Disks:    make(map[string]string),
+	}
+}
+
+func (d *DiskExtractor) Write(m Machine) error {
+	labelKey := m.Labels.String()
+	if _, ok := d.Selector[labelKey]; ok {
+		d.Disks[labelKey] = m.Disk
+	}
+	return nil
+}
+
+func (d *DiskExtractor) RegisterHardwareSelector(hardwareSelector eksav1alpha1.HardwareSelector) {
+	key := serializedHardwareSelector(hardwareSelector)
+	if _, ok := d.Selector[key]; !ok {
+		d.Selector[key] = hardwareSelector
+	}
+}
+
+func serializedHardwareSelector(hardwareSelector eksav1alpha1.HardwareSelector) string {
+	selectorKey := make(Labels)
+	for key, val := range hardwareSelector {
+		selectorKey[key] = val
+	}
+	return selectorKey.String()
+}
+
+func (d *DiskExtractor) GetDisk(hardwareSelector eksav1alpha1.HardwareSelector) (string, error) {
+	key := serializedHardwareSelector(hardwareSelector)
+	if _, ok := d.Disks[key]; !ok {
+		return "", fmt.Errorf("hardware selector does not match with any given hardware machines %v", hardwareSelector)
```

Comment:
Use colon's in errors to separate added context from original error.

```suggestion
		return "", fmt.Errorf("hardware selector does not match with any given hardware machines: %v", hardwareSelector)
```